src/model/pitch.rs

```diff
@@ -0,0 +1,51 @@
+use crate::{Letter, Modifier};
+use combine::{Parser, Stream};
+use std::fmt;
+use strum::IntoEnumIterator;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Pitch {
+    pub letter: Letter,
+    pub modifier: Modifier,
+}
+
+impl Pitch {
+    pub fn new(letter: Letter, modifier: Modifier) -> Self {
+        Self { letter, modifier }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        (Letter::parser(), Modifier::parser()).map(|(letter, modifier)| Self::new(letter, modifier))
+    }
+
+    pub fn all() -> Vec<Pitch> {
+        Letter::iter()
+            .flat_map(|l| Modifier::iter().map(move |m| Pitch::new(l, m)))
```

Comment:
Won't this generate eg `Fb`?

Reply:
Yes, but isn't that fine? `Fb` is a real pitch.

Reply:
It seems unintuitive as far as what "iterating over all the pitches" means, I'd expect that to be 12 items long (yes especially when thinking about adding `#`'s then there's a difference between "all pitch names" vs "all unique pitch names" I guess?)

Reply:
Ah, gotcha. I guess I'd imagine `Pitches::in_key(...)` doing something like what you're describing.

Reply:
Ya I guess it's from the perspective of the consumer that it seems weird - what is one to understand conceptually that `Pitch::all()` gives you? Currently it's just used for enumerating test cases, and it's hard to picture it being used for anything else because of that lack of conceptual meaning. But ya this is pointing at interesting questions around the musical meaning of these as names vs actual things, eg once `#`'s are added (or technically already with eg `Fb` vs `E`) there'll presumably at least eventually need to be some concept of aliasing, so I guess I start to picture 12 canonical "pitch instances" (enum values?) each of which has one or more "aliases"?

And to throw another wrinkle, thinking about `Fb` being valid, I'd say so is eg `Abb` in the same sense (the flat 5 in the key of Db). Again this feels sort of like an aliasing concept but one that'd we'll probably eventually run into if we go very far down the road of generating things based on a programmatic understanding of theory

Reply:
Maybe this should be moved into the test mod. I think I'm thinking about this backwards from you. I'd say that there are many pitches, some of which happen to have the same frequency representation (on equal-tempered instruments, and MIDI), but that they are functionally different things.